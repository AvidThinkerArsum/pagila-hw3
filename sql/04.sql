/*
 * List the first and last names of all actors who:
 * 1. have appeared in at least one movie in the "Children" category,
 * 2. but that have never appeared in any movie in the "Horror" category.
 */

SELECT DISTINCT first_name, last_name FROM actor
JOIN film_actor USING (actor_id)
JOIN film ON film_actor.film_id = film.film_id
JOIN film_category ON film.film_id = film_category.film_id
JOIN category USING (category_id)
WHERE actor.actor_id IN
(SELECT actor.actor_id FROM actor
JOIN film_actor USING (actor_id)
JOIN film ON film_actor.film_id = film.film_id
JOIN film_category ON film.film_id = film_category.film_id
JOIN category USING (category_id)
WHERE name = 'Children')
AND actor.actor_id NOT IN
(SELECT actor.actor_id FROM actor
JOIN film_actor USING (actor_id)
JOIN film ON film_actor.film_id = film.film_id                 
JOIN film_category on film.film_id = film_category.film_id
JOIN category USING (category_id)
WHERE name = 'Horror')
ORDER BY last_name;
