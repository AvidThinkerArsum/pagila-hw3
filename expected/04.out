 first_name | last_name 
------------+-----------
 JUDY       | DEAN
 SPENCER    | DEPP
 AL         | GARLAND
 THORA      | TEMPLE
(4 rows)

